bundles/org.openhab.binding.linky/src/main/java/org/openhab/binding/linky/internal/handler/LinkyHandler.java

```diff
@@ -237,26 +246,28 @@ private synchronized void updateMonthlyData() {
      */
     private synchronized void updateYearlyData() {
         if (isLinked(LAST_YEAR) || isLinked(THIS_YEAR)) {
-            Consumption result = cachedYearlyData.getValue();
-            if (result != null) {
-                Aggregate years = result.aggregats.years;
+            cachedYearlyData.getValue().ifPresent(values -> {
+                Aggregate years = values.aggregats.years;
                 updateKwhChannel(LAST_YEAR, years.datas.get(0));
-                updateKwhChannel(THIS_YEAR, years.datas.get(1));
-            }
+                if (years.datas.size() > 1) {
+                    updateKwhChannel(THIS_YEAR, years.datas.get(1));
+                } else {
+                    updateKwhChannel(LAST_YEAR, Double.NaN);
```

Comment:
I assume this is supposed to be:
```suggestion
                    updateKwhChannel(THIS_YEAR, Double.NaN);
```

Reply:
Ooops, nicely found